src/main/java/frc/robot/vision/VisionFilters.java

```diff
@@ -0,0 +1,65 @@
+package frc.robot.vision;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.constants.Field;
+import frc.constants.MathConstants;
+import frc.robot.vision.data.AprilTagVisionData;
+import frc.robot.vision.data.VisionData;
+import frc.utils.Filter;
+import frc.utils.ToleranceUtils;
+
+public class VisionFilters {
+
+	public static Filter<VisionData> isPitchAtAngle(Rotation2d wantedPitch, Rotation2d pitchTolerance) {
+		return new Filter<>(
+			visionData -> MathUtil.isNear(
+				wantedPitch.getRadians(),
+				visionData.getEstimatedPose().getRotation().getY(),
+				pitchTolerance.getRadians(),
+				0,
+				MathConstants.FULL_CIRCLE.getRadians()
```

Comment:
maybe create in tolernace utils wrappedIsNear that is doint the min and max for you.